code/src/components/app-range-slider/AppRangeSlider.tsx

```diff
@@ -0,0 +1,97 @@
+import { ChangeEvent, useMemo, useState } from "react";
+
+import Slider from "@mui/material/Slider";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppInput from "@/components/app-input/AppInput";
+import { AppRangeSliderProps } from "@/components/app-range-slider/AppRangeSlider.types";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import cn from "@/utils/cn/cn";
+
+import "@/components/app-range-slider/AppRangeSlider.scss";
+
+const AppRangeSlider = ({
+  className,
+  onChange,
+  min = 0,
+  max = 20000,
+  step = 10,
+  value,
+  ...props
+}: AppRangeSliderProps) => {
+  const [rangeStart, setRangeStart] = useState(value?.[0] ?? min);
+  const [rangeEnd, setRangeEnd] = useState(value?.[1] ?? max);
+
+  const commonRangeProps = useMemo(
+    () => ({ min, max, step }),
+    [min, max, step]
+  );
```

Comment:
how this `useMemo` memoization works? If we not defined even dependencies.
Can record a little video before and after usage. Just interesting how it works here

Reply:
I don't see any sense of recording a little video for it. I used `useMemo` here because `rangeStart`, `rangeEnd` and `value` states change often, but `min`, `max` ans `step` props are supposed to change rarely, most likely on initial setup, so I tried to save extra performance here

Reply:
> I don't see any sense of recording a little video for it. I used `useMemo` here because `rangeStart`, `rangeEnd` and `value` states change often, but `min`, `max` ans `step` props are supposed to change rarely, most likely on initial setup, so I tried to save extra performance here

I just want to see the differences using of it.
Whats heppen if we not use memoization ? Please explain it in video. but for me it's not understandable

Reply:
I brainstormed that we can remove it for now, because in this case it is not a big deal